src/Donations/Micros/DonationTypes/BouquetDonations.tsx

```diff
@@ -69,9 +69,15 @@ function FundingDonations({ setopenCurrencyModal }: Props): ReactElement {
       for (const option of paymentSetup.options) {
         newallOptionsArray.push(option.quantity);
       }
+      const defaultPaymentOption = paymentSetup.options.filter(
+        (option) => option.isDefault === true
+      );
       const newQuantity = router.query.units
         ? Number(router.query.units)
+        : defaultPaymentOption.length > 0
+        ? defaultPaymentOption[0].quantity
         : paymentSetup.options[1].quantity;
+
       if (newQuantity && !newallOptionsArray.includes(newQuantity)) {
         setCustomInputValue(
           paymentSetup.unitBased ? quantity * paymentSetup.unitCost : quantity
```

Comment:
Why using `quantity` instead of `newQuantity` here?

Reply:
Also why using `setquantity(Number(router.query.units));` instead of `setquantity(Number(newQuantity));` in the line below?